### ISSUE TITLE
Consider adding `is.shiny.tag()` function

### ISSUE THREAD

**Issue by daattali** [rstudio/htmltools#389]
It's fairly common for me, I'm not sure about others, to want to know whether or not a particular object is a shiny tag. This is usually done as error checking arguments in a function.

I often add the following function to my utils file:

```
is.shiny.tag <- function(x) {
  inherits(x, "shiny.tag")
}
```

Would you accept a PR to add this to {htmltools}, or would you prefer not?

**Comment by cpsievert** [comment]
I agree this is a pain, and would love a PR. Seems like it'd also be useful to have similar `is*()` functions for `tagList()`, `HTML()`, `HTMLDependency()`?

**Comment by daattali** [comment]
I can help with tag and tagList. For HTML/dependency I don't feel comfortable writing their `is` function and their tests. 

For tagList, should it be `is.tagList`?
